### ISSUE TITLE
dlicd is not working on AlpineLinux

### ISSUE THREAD

**Issue by joker85** [50ButtonsEach/fliclib-linux-hci#69]
Hello, 

I have a problem to run flicd on my Raspberry Pi 3 

0cd90890fd17:/etc/conf.d# cd /root/fliclib-linux-hci-master/bin/armv6l/
0cd90890fd17:~/fliclib-linux-hci-master/bin/armv6l# ls
flicd
0cd90890fd17:~/fliclib-linux-hci-master/bin/armv6l# ./flicd -f flic.sqlite3 -l ./flic.out
-ash: ./flicd: not found
0cd90890fd17:/addons/flicd# /bin/bash ./flicd -f pippo
./flicd: ./flicd: cannot execute binary file
0cd90890fd17:~/fliclib-linux-hci-master/bin/armv6l# cat /etc/os-release
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.6.1
PRETTY_NAME="Alpine Linux v3.6"
HOME_URL="http://alpinelinux.org"
BUG_REPORT_URL=“http://bugs.alpinelinux.org”

Is the binary file incompatible with my of version or I'm doing something wrong?

I had this issue with the latest version of home-assistant.

Thanks for your support.

**Comment by Emill** [comment]
What does lscpu print?

**Comment by joker85** [comment]
Hi,

# lscpu 
Architecture:          armv7l
Byte Order:            Little Endian
CPU(s):                4
On-line CPU(s) list:   0-3
Thread(s) per core:    1
Core(s) per socket:    4
Socket(s):             1
Model name:            ARMv7 Processor rev 4 (v7l)
CPU max MHz:           1200.0000
CPU min MHz:           600.0000

**Comment by Emill** [comment]
Hmm not really sure why this happens. Could you try use the full path instead of ./

**Comment by Emill** [comment]
And Exactly which download did you pick when you downloaded AlpineLinux?

**Comment by joker85** [comment]
I have a build created by home assistant: https://home-assistant.io/hassio/installation/

The build allow me just to run application in docker container, maybe it is necessary to add some specific components in the container.

**Comment by Emill** [comment]
Not sure how easy it is to forward the hci device to the docker environment. Do you really need to use Docker?

**Comment by joker85** [comment]
I have already done the docker container to run it, but I'm not sure if the problem is the Docker container or the OS.

If the software need some specific library or other software to work I can handle it, but i need to know the list.

**Comment by Emill** [comment]
It doesn't need anything else than a Linux kernel, libc, network connection and Bluetooth hci device.

Could you run "file /bin/bash" and print the output. Or "file [some other binary]"? It should print a lot of things about the cpu and binary format.

**Comment by acarlo79** [comment]
I have the same issue, it looks like that the Raspberry binary is not compatible with this distribution:
The binary is not recognised as such.

here the output requested:

/bin/bash: ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-musl-armhf.so.1, stripped, with debug_info

**Comment by Emill** [comment]
Maybe it is because that distribution uses musl instead of glibc...

**Comment by acarlo79** [comment]
3b0323706b6a:/config# /bin/bash /config/flicd -f db.sql
/config/flicd: /config/flicd: cannot execute binary file
3b0323706b6a:/config# ls -la flicd 
-rwxrwxrwx    1 root     root        734708 Jul 29 21:53 flicd
3b0323706b6a:/config# file flicd 
flicd: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-armhf.so.3, for GNU/Linux 2.6.32, BuildID[sha1]=e080a631d9d667cba75284641c9a01c16cffdc8a, stripped, with debug_info
3b0323706b6a:/config# file /bin/bash
/bin/bash: ELF 32-bit LSB shared object, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-musl-armhf.so.1, stripped, with debug_info
3b0323706b6a:/config#

**Comment by acarlo79** [comment]
well, possibly as you said, the only fix is to get a version compiled for using musl.

**Comment by Emill** [comment]
Yes but that will take some time. I'm back from holiday in August.

**Comment by acarlo79** [comment]
we can just wait :)

**Comment by joker85** [comment]
Yes perfect thanks.

**Comment by joker85** [comment]
Hello,

have you got the time to compile the new binary?

Thanks.
